bids/layout/tests/test_layout.py

```diff
@@ -48,6 +48,17 @@ def test_index_metadata(index_metadata, query, result, mock_config):
     assert metadata.get('RepetitionTime') == result
 
 
+
+@pytest.mark.parametrize('config_type', [str, Path])
+def test_config_filename(config_type):
+    data_path = Path(get_test_data_path())
+    # Use custom config that replaces session with oligarchy
+    config_path = data_path.parent / 'bids_specs_with_oligarchy.json'
+    layout = BIDSLayout(data_path / "7t_trt", config=config_type(config_path))
```

Comment:
@kousu Note that I had to pass `config` to `BIDSLayout`, not `BIDSLayoutIndexer`. This section will pass `None` to `ConnectionManager`, which will default to `'bids'`.

https://github.com/bids-standard/pybids/blob/23553339dda7537752ace92d6c99a3f523e06ee2/bids/layout/layout.py#L137-L141

The `BIDSLayoutIndexer.config_filename` is for where to look inside the dataset for an additional configuration file. This is used for a pipeline that wants to generate derivatives with new entities. PyBIDS will then pick that additional config up (or should).